dev/tests/js/jasmine/tests/app/code/Magento/AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/overlay.test.js

```diff
@@ -0,0 +1,147 @@
+            });
+
+            it('Check method type', function () {
+                var type = typeof overlayObj.updateLicensed;
+                expect(type).toEqual('function');
+            });
+
+            it("Check Ajax Success request", function() {
+                var login = {
+                    user: function () {
+                        return {
+                            isAuthorized: true
+                        };
+                    }
+                }, ids = [1,2,3];
+                spyOn(overlayObj, 'login').and.callFake(function(e) {
+                    return login;
+                });
+                spyOn(overlayObj, 'getIds').and.callFake(function(e) {
+                    return ids;
+                });
+                spyOn($, 'ajax').and.callFake(function(e) {
+                    return $.Deferred().resolve({
+                        'response': 'success'
+                    }).promise();
+                });
+                overlayObj.updateLicensed();
+                expect(overlayObj.updateLicensed()).toBeUndefined();
+                expect($.ajax).toHaveBeenCalled();
+                expect(overlayObj.getIds).toHaveBeenCalled();
+            });
+
+            it('Check Ajax failure request', function() {
+                var login = {
+                    user: function () {
+                        return {
+                            isAuthorized: false
+                        };
+                    }
+                }, ids = [1,2,3];
+                spyOn(overlayObj, 'login').and.callFake(function(e) {
+                    return login;
+                });
+                spyOn(overlayObj, 'getIds').and.callFake(function(e) {
+                    return ids;
+                });
+                spyOn($, 'ajax');
+                overlayObj.updateLicensed();
+                expect(overlayObj.updateLicensed()).toBeUndefined();
+                expect($.ajax).not.toHaveBeenCalled();
+                expect(overlayObj.getIds).not.toHaveBeenCalled();
+            });
+        });
+
+        describe('"getIds" method', function () {
+
+            it('Check for defined ', function () {
+                expect(overlayObj.hasOwnProperty('getIds')).toBeDefined();
+            });
+
+            it('Check method type', function () {
+                var type = typeof overlayObj.getIds;
+                expect(type).toEqual('function');
+            });
+
+        });
+
+        describe('"getStyles" method', function () {
+
+            it('Check for defined ', function () {
+                expect(overlayObj.hasOwnProperty('getStyles')).toBeDefined();
+            });
+
+            it('Check method type', function () {
+                var type = typeof overlayObj.getStyles;
+                expect(type).toEqual('function');
+            });
+
+        });
+
+        describe('"isVisible" method', function () {
```

Comment:
Once more, we should add logical tests here. Since `isVisible` returns either `true` or `false`, I would write  two additional tests  for this method: when does it return `true`, and when does it return `false`?